app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -44,55 +45,67 @@ def identify_duplicates
       # If we do find a match:
       #   create the associated WarehouseClient
 
-      Rails.logger.info "Matching #{unprocessed.count} unprocessed clients"
+      unprocessed_count = unprocessed.count
+      Rails.logger.info 'Matching #unprocessed_count} unprocessed clients'
       matched = 0
       new_created = 0
+
+      # Find all potential matches
+      more_than_one_match = find_merge_candidates_for_unprocessed
+      found_ids = more_than_one_match.flatten.uniq.to_set
+
+      # At this point more_than_one_match contains pairs of destination client id and source client id where the
+      # source client should be merged into the destination client.
+      # If the source client does not exist in any of the pairs, a new destination client should be created for it.
+      destination_clients_by_id = GrdaWarehouse::Hud::Client.where(id: found_ids.to_a).index_by(&:id)
+
       unprocessed.find_in_batches do |batch|
         matched_ids = []
         destination_client_updates = []
         new_destination_clients = []
         new_warehouse_clients = {}
         source_client_ids_with_new_destination_clients = []
+
         batch.each do |client|
-          match_id = check_for_obvious_match(client)
-          if match_id.present?
-            matched += 1
```

Comment:
If we remove this then it looks like matched isn't incremented anymore? That seems incorrect